File: Python/Labs/Module 3/Lab 1/Q2 - Cookies/main.py
tar=int(input())
arr=input().split()
a=[]
count=0
for i in arr:
    a.append(int(i))
for i in range(0,len(a)-1):
    ch=[]
    for j in a:
        if j>tar:
            ch.append(True)
        else:
            ch.append(False)
    if False in ch:
        a.sort()
        ele=a[0]+2*a[1]
        a.pop(0)
        a.pop(0)
        a.insert(0,ele)
        count+=1
    else:
        break
print(count)
                                                                                                                            

File: Java/Labs/Module 3/Lab 1/Q2 - Cookies/Main.java
import java.util.*;
public class Main {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int tar = input.nextInt();
        input.nextLine();
        String[] arr = input.nextLine().split(" ");
        int count=0;
        int[] a=new int[arr.length];
        for(int i=0;i<arr.length;i++){
            a[i]=Integer.parseInt(arr[i]);
        }
        for(int i=0;i<a.length-1;i++){
            boolean[] ch=new boolean[arr.length];
            int c=0;
            for(int j:a){
                if(j>tar){
                    ch[c]=true;
                } else{
                    ch[c]=false;
                }
                c++;
            }
            boolean check=true;
            for(boolean y:ch){
                if(!y){
                    check=false;
                }
            }
            if(!check){
                Arrays.sort(a);
                int ele=a[0]+2*a[1];
                a=Arrays.copyOfRange(a,1,a.length);
                a[0]=ele;
                count++;
            } else{
                break;
            }
        }
        System.out.println(count+1);
    }
}
                                                                                                                            